From f43e00ff22b6c19195d0d2b9905683413464cc07 Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@st.com>
Date: Wed, 15 Feb 2023 17:17:53 +0100
Subject: [PATCH 0268/1141] arm64: dts: st: add FDCAN1/FDCAN3 support on
 stm32mp25

Update the device-tree to add support for FDCAN
FDCAN modules are compliant with ISO 11898-1: 2015
and CAN FD protocol specification version 1.0.
In addition, the first CAN module FDCAN1 supports time
triggered CAN (TTCAN), specified in ISO 11898-4, including
event synchronized time-triggered communication, global system
time, and clock drift compensation. The FDCAN1 contains
additional registers, specific to the time triggered feature.

Signed-off-by: Christophe Roullier <christophe.roullier@st.com>
Change-Id: I715eae50a00f0cf3c80c5dfa9658b131575905c5
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/291747
Tested-by: Eric FOURMONT <eric.fourmont-ext@st.com>
Reviewed-by: Eric FOURMONT <eric.fourmont-ext@st.com>
Domain-Review: Eric FOURMONT <eric.fourmont-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/310722
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Alexandre TORGUE <alexandre.torgue@foss.st.com>
Domain-Review: Alexandre TORGUE <alexandre.torgue@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp253.dtsi | 28 ++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp253.dtsi b/arch/arm64/boot/dts/st/stm32mp253.dtsi
index bb75164b3fee..66a32b070d89 100644
--- a/arch/arm64/boot/dts/st/stm32mp253.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp253.dtsi
@@ -25,6 +25,34 @@ arm-pmu {
 
 	soc@0 {
 		rifsc: rifsc@42080000 {
+			m_can1: can@402d0000 {
+				compatible = "bosch,m_can";
+				reg = <0x402d0000 0x400>, <0x40310000 0x1400>;
+				reg-names = "m_can", "message_ram";
+				interrupts = <GIC_SPI 91 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 94 IRQ_TYPE_LEVEL_HIGH>;
+				interrupt-names = "int0", "int1";
+				clocks = <&rcc CK_BUS_FDCAN>, <&rcc CK_KER_FDCAN>;
+				clock-names = "hclk", "cclk";
+				bosch,mram-cfg = <0x0 0 0 32 0 0 2 2>;
+				feature-domains = <&rifsc STM32MP25_RIFSC_FDCAN_ID>;
+				status = "disabled";
+			};
+
+			m_can3: can@402f0000 {
+				compatible = "bosch,m_can";
+				reg = <0x402f0000 0x400>, <0x40310000 0x2800>;
+				reg-names = "m_can", "message_ram";
+				interrupts = <GIC_SPI 93 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 96 IRQ_TYPE_LEVEL_HIGH>;
+				interrupt-names = "int0", "int1";
+				clocks = <&rcc CK_BUS_FDCAN>, <&rcc CK_KER_FDCAN>;
+				clock-names = "hclk", "cclk";
+				bosch,mram-cfg = <0x1400 0 0 32 0 0 2 2>;
+				feature-domains = <&rifsc STM32MP25_RIFSC_FDCAN_ID>;
+				status = "disabled";
+			};
+
 			eth2: eth2@482d0000 {
 				compatible = "st,stm32mp25-dwmac", "snps,dwmac-5.10a";
 				reg = <0x482d0000 0x4000>;
-- 
2.39.2

